x/finality/keeper/gov.go

```diff
@@ -41,28 +39,56 @@ func (k Keeper) HandleResumeFinalityProposal(ctx sdk.Context, fpPksHex []string,
 			return fmt.Errorf("the finality provider %s has voted for height %d", fpPkHex, haltingHeight)
 		}
 
+		fpBtcPk := fpPk.MustMarshal()
+		fp, err := k.BTCStakingKeeper.GetFinalityProvider(ctx, fpBtcPk)
+		if err != nil {
+			return fmt.Errorf("failed to find the finality provider %s in btcstaking: %w", fpPkHex, err)
+		}
+
+		k.Logger(ctx).Debug(
+			"fp running proposal resume finality",
+			"jailed", fp.IsJailed(),
+			"slashed", fp.IsSlashed(),
+			"height", haltingHeight,
+			"public_key", fpPkHex,
+		)
+
+		// if the FP is already jailed or slashed, no need to try to set to jail
+		// or to update the signing info
+		if fp.IsSlashed() || fp.IsJailed() {
+			continue
+		}
```

Comment:
If the fp is already jailed or slashed, we should still set their voting power at the halting height `0`, right? For example, a fp is slashed/jailed at height `200` but the halting height is `100`, then when we execute the proposal, we should still set the fp's voting power to 0 from height `100`

Reply:
We are doing this, check this test https://github.com/babylonlabs-io/babylon/blob/c801d4d5606c8a539cae20a7c1097a56ad0dc0ab/test/replay/finality_test.go#L45